### PR TITLE
[for test]executor, expression: support chunk for hashjoin(single thread)

### PR DESCRIPTION
**Pull request by XuHuaiyu**

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
PTAL @coocood @zz-jason

File: executor/join.go [hunk 1]
```diff
@@ -446,6 +509,109 @@ func (e *HashJoinExec) Next(goCtx goctx.Context) (Row, error) {
...
+	}
+	return nil
+}
+
+func (e *HashJoinExec) join2Chunk(outerChk *chunk.Chunk) (err error) {
+	e.outerSelected, err = expression.VectorizedFilter(e.ctx, e.outerFilter, outerChk, e.outerSelected)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	e.resultChunk = e.newChunk()
```
Comment (coocood) [inline executor/join.go]: This can be reused.

File: executor/join.go [hunk 2]
```diff
@@ -446,6 +509,109 @@ func (e *HashJoinExec) Next(goCtx goctx.Context) (Row, error) {
...
+	}
+	innerPtrs := e.hashTable.Get(joinKey)
+	if len(innerPtrs) == 0 {
+		err = e.resultGenerator.emitToChunk(outerRow, nil, chk)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		return nil
+	}
+	innerRows := make([]chunk.Row, 0, len(innerPtrs))
```
Comment (coocood) [inline executor/join.go]: This memory can be reused.

File: executor/join.go [hunk 3]
```diff
@@ -164,6 +174,27 @@ func getJoinKey(sc *stmtctx.StatementContext, cols []*expression.Column, row Row
 	return false, bytes, errors.Trace(err)
 }
 
+func (e *HashJoinExec) getJoinKeyFromChkRow(keys []*expression.Column, row chunk.Row, keyBuf []byte) (hasNull bool, _ []byte, err error) {
+	keyColIdx := make([]int, len(keys))
```
Comment (coocood) [inline executor/join.go]: This allocation can be avoided.

File: executor/join.go [hunk 4]
```diff
@@ -164,6 +174,27 @@ func getJoinKey(sc *stmtctx.StatementContext, cols []*expression.Column, row Row
...
+	keyColTypes := make([]*types.FieldType, len(keys))
```
Comment (coocood) [inline executor/join.go]: ditto

**Pull request closed without merging**